hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java

```diff
@@ -53,17 +78,28 @@
      * The iterator may be requested only once.
      *
      * @return iterator
-     * @throws IllegalStateException if the method is invoked more than once
+     * @throws IllegalStateException if the method is invoked more than once or
+     *    if this result doesn't have rows (i.e. when {@link #isUpdateCount()}
+     *    returns {@code true})
      * @throws SqlException in case of an SQL-related error condition
      */
     @Nonnull
     @Override
     Iterator<SqlRow> iterator();
 
     /**
-     * Release the resources associated with the query result.
+     * Returns the number of rows updated by the execution.
+     *
+     * @throws IllegalStateException if this result doesn't represent an update
+     *     count (i.e. when {@link #isUpdateCount()} returns {@code false})
+     */
+    long updatedCount();
```

Comment:
There is an inconsistency in naming: `updatedCount`, but is `isUpdateCount`. 
Should we rename this method to JDBC-like `updateCount`?

Reply:
agree